packages/medusa/src/repositories/product.ts

```diff
@@ -314,4 +323,60 @@ export class ProductRepository extends Repository<Product> {
 
     return this.findByIds(productIds)
   }
+
+  public async getFreeTextSearchResultsAndCount(
+    q: string,
+    options: CustomOptions = { where: {} },
+    relations: (keyof Product)[] = []
+  ): Promise<[Product[], number]> {
+    const cleanedOptions = this._cleanOptions(options)
+
+    let qb = this.createQueryBuilder("product")
+      .leftJoinAndSelect("product.variants", "variant")
+      .leftJoinAndSelect("product.collection", "collection")
+      .select(["product.id"])
+      .where(cleanedOptions.where)
+      .andWhere(
+        new Brackets((qb) => {
+          qb.where(`product.description ILIKE :q`, { q: `%${q}%` })
+            .orWhere(`product.title ILIKE :q`, { q: `%${q}%` })
+            .orWhere(`variant.title ILIKE :q`, { q: `%${q}%` })
+            .orWhere(`variant.sku ILIKE :q`, { q: `%${q}%` })
+            .orWhere(`collection.title ILIKE :q`, { q: `%${q}%` })
+        })
+      )
+      .skip(cleanedOptions.skip)
+      .take(cleanedOptions.take)
+
+    if (cleanedOptions.withDeleted) {
+      qb = qb.withDeleted()
+    }
+
+    const [results, count] = await qb.getManyAndCount()
+
+    const products = await this.findWithRelations(
+      relations,
+      results.map((r) => r.id),
+      cleanedOptions.withDeleted
+    )
+
+    return [products, count]
+  }
+
+  private _cleanOptions(
```

Comment:
comment: As suggested above, not sure that method should exists, at least at the moment and with that impl.

Reply:
hmm .. I tried giving this some thought and I feel like there is not really a disadvantage with having the method in the repo class. In fact, one might even argue that it's better because reading the `getFreeTextSearchResultsAndCount` method becomes a tiny bit easier to read. When reading the method, you learn that we're cleaning the options without paying too much attention to the specifics of what it does (which can be a bit distracting if left inside the `getFreeTextSearchResultsAndCount` method). 

Ofc, I don't think this is too important and I'm okay with removing the method, just thought I'd share my thoughts here in case you have some counter arguments 😄

Reply:
The thing is that it is equivalent at 
delete where.description 
delete where.title

which I think is still readable and self explanatory.
This is why i thought for the moment this method does not add a lot of value in term of
Separation nor understanding.

But in the end, i can be fine with both😂